atc/db/resource_type.go

```diff
@@ -215,6 +226,23 @@ func (t *resourceType) Reload() (bool, error) {
 	return true, nil
 }
 
+func (r *resourceType) SetResourceConfigScope(scope ResourceConfigScope) error {
+	_, err := psql.Update("resource_types").
+		Set("resource_config_id", scope.ResourceConfig().ID()).
+		Where(sq.Eq{"id": r.id}).
+		Where(sq.Or{
+			sq.Eq{"resource_config_id": nil},
+			sq.NotEq{"resource_config_id": scope.ResourceConfig().ID()},
+		}).
```

Comment:
This is just to avoid performing an unnecessary write when the scope is already set to the same value. The `sq.Eq` `NULL` check is probably redundant with the `sq.NotEq` but it felt a little clearer to check for it explicitly, since `NULL` is the initial state.